test/jasmine/tests/plot_api_react_test.js

```diff
@@ -886,3 +887,490 @@ describe('resizing with Plotly.relayout and Plotly.react', function() {
         .then(done);
     });
 });
+
+
+describe('Plotly.react and uirevision attributes', function() {
```

Comment:
I didn't write tests for every possible GUI-editable attribute yet, it's probably worthwhile doing that though. The missing ones:

- [ ] ternary zoom/pan
- [ ] polar zoom/pan & rotation
- [ ] mapbox zoom/pan
- [ ] parcoords `constraintrange`
- [ ] `editable: true`: shape positions, annotation positions & text, plot, axis, & colorbar titles, colorbar & legend positions

Reply:
Heh yes, definitely worthwhile - quite a few of them exposed broken cases, either in my new code or in their existing code. AFAIK they're all covered now.